bridge/server/services/data-service.ts

```diff
@@ -245,4 +276,50 @@ export class DataService {
   private buildRemediationEvent(stageName: string): string {
     return `sh.keptn.event.${stageName}.${SequenceTypes.REMEDIATION}.${EventState.TRIGGERED}`;
   }
+
+  private _getResourceFileTree(resources: Resource[]): TreeEntry[] {
+    const directory: TreeDirectory = {_: []};
+
+    for (const res of resources) {
+      const parts = res.resourceURI.split('/').filter(item => !!item);
+      this._addToTreeDirectory(directory, parts);
+    }
+
+    return this._buildTree(directory, '').children || [];
+  }
+
+  private _addToTreeDirectory(currentDirectory: TreeDirectory, parts: string[]): void {
+    let index = 0;
+    for (; index < parts.length - 1; ++index) {
+      const part = parts[index];
+      // @ts-ignore
+      currentDirectory[part] ??= {_: []};
+      // @ts-ignore
+      currentDirectory = currentDirectory[part];
+    }
+    currentDirectory._.push(parts[index]);
+  }
+
+  private _buildTree(currentDirectory: TreeDirectory, fileName: string): TreeEntry {
+    const tree: TreeEntry = {fileName, children: [] as TreeEntry[]};
+    const dict: { [key: string]: boolean } = {_: true};
+    const folders: TreeEntry[] = [];
+    const files: TreeEntry[] = [];
+    for (const key in currentDirectory) {
+      if (dict[key]) {
+        files.push(...currentDirectory._.map(item => {
+          return {
+            fileName: item,
+          } as TreeEntry;
```

Comment:
If we add `children: []` here, we don't need to consider this case client-side.

Reply:
We leave it as is :)